text/1003-deprecation-import-ember-from-ember.md

```diff
@@ -0,0 +1,344 @@
+- 🌐 `Ember.Test.QUnitAdapter`
+- 🌐 `Ember.setupForTesting`
+
+These will need to be moved to a module such as `@ember/test`.
+
+### A way to communicate with the ember-inspector
+
+The inspector will be hit especially hard by the removal of these APIs.
+
+A good few already have available imports though.
+
+|   | API | import |
+| - | --- | ------ |
+|🔒| `Ember.meta` | `import { meta } from '@ember/-internals/meta';` |
+|🌐| `Ember.VERSION` | none |
+|🔒| `Ember._captureRenderTree` | `import { captureRenderTree } from '@ember/debug';` |
+|🔒| `Ember.instrument` | `import { instrument } from '@ember/instrumentation';` |
+|🔒| `Ember.subscribe` | `import { subscribe } from '@ember/instrumentation';` |
+|🔒| `Ember.Instrumentation.*` | `import { * } from '@ember/instrumentation';`  |
+|🫣| `Ember.ViewUtils` | `import * as viewUtils from '@ember/-internals/views';`[^view-utils]  |
+|🔒| `Ember.ViewUtils.getChildViews` | `import { getChildViews } from '@ember/-internals/views';` |
+|🫣| `Ember.ViewUtils.getElementView` | `import { getElementView } from '@ember/-internals/views';` |
+|🔒| `Ember.ViewUtils.getRootViews` | `import { getRootViews } from '@ember/-internals/views';` |
+|🔒| `Ember.ViewUtils.getViewBounds` | `import { getViewBounds } from '@ember/-internals/views';` |
+|🔒| `Ember.ViewUtils.getViewBoundingClientRect` | `import { getViewBoundingClientRect } from '@ember/-internals/views';` |
+|🔒| `Ember.ViewUtils.getViewClientRects` | `import { getViewClientRects } from '@ember/-internals/views';` |
+|🔒| `Ember.ViewUtils.getViewElement`  | `import { getViewElement } from '@ember/-internals/views';` |
+|🫣| `Ember.ViewUtils.isSimpleClick` | `import { isSimpleClick } from '@ember/-internals/views';` |
+|🫣| `Ember.ViewUtils.isSerializationFirstNode` | `import { isSerializationFirstNode } from '@ember/-internals/glimmer';` |
+
+[^view-utils]: Not all of these exports are used for `ViewUtils`.
+
+
+Perhaps we can have folks add this to their apps:
+```js
+import { macroCondition, isDevelopingApp, importSync } from '@embroider/macros';
+
+if (macroCondition(isDevelopingApp())) {
+  // maybe this is side-effecting and installs 
+  // some functions on `globalThis` that the inspector could call
+  importSync('@ember/inspector-support');
+}
+```
+
+### No replacements.
+
+Applies to both the value and type exports (if applicable).
+
+- 🫣 `Ember._getPath`
+- 🫣 `Ember.isNamespace`
+- 🫣 `Ember.toString`
+- 🔒 `Ember.Container`
+- 🔒 `Ember.Registry`
+
+Internal decorator utils
+- 🫣 `Ember._descriptor`
+- 🔒 `Ember._setClassicDecorator`
+
+Reactivity
+- 🔒 `Ember.beginPropertyChanges`
+- 🔒 `Ember.changeProperties`
+- 🔒 `Ember.endPropertyChanges`
+
+Observable 
+- 🌐 `Ember.hasListeners`
+
+Mixins
+- 🔒 `Ember._ContainerProxyMixin`
+- 🔒 `Ember._ProxyMixin`
+- 🔒 `Ember._RegistryProxyMixin`
+- 🔒 `Ember.ActionHandler`
+- 🔒 `Ember.Comparable`
+
+Utility
+- 🫣 `Ember.lookup`
+- 🌐 `Ember.libraries` - 
```

Comment:
Here is a replacement for not only libraries, but _everything_ from the build environment:


https://github.com/NullVoxPopuli/embroider-with-webpack-build-plugins-demo


Uses: https://github.com/yjl9903/unplugin-info/tree/main

My code: 
```js
import * as info from '~build/package';
import * as meta from '~build/meta';

console.log(JSON.stringify({ info, meta }, null, 2));
```

<details><summary>Output</summary>

```
 {
  "info": {
    "author": "",
    "dependencies": {
      "unplugin-info": "^1.1.0"
    },
    "description": "Small description for embroider-with-webpack-build-plugins-demo goes here",
    "devDependencies": {
      "@babel/core": "^7.23.7",
      "@babel/eslint-parser": "^7.23.3",
      "@babel/plugin-proposal-decorators": "^7.23.7",
      "@ember/optional-features": "^2.0.0",
      "@ember/string": "^3.1.1",
      "@ember/test-helpers": "^3.2.1",
      "@embroider/compat": "^3.4.3",
      "@embroider/core": "^3.4.3",
      "@embroider/webpack": "^3.2.1",
      "@glimmer/component": "^1.1.2",
      "@glimmer/tracking": "^1.1.2",
      "broccoli-asset-rev": "^3.0.0",
      "concurrently": "^8.2.2",
      "ember-auto-import": "^2.7.2",
      "ember-cli": "~5.6.0",
      "ember-cli-app-version": "^6.0.1",
      "ember-cli-babel": "^8.2.0",
      "ember-cli-clean-css": "^3.0.0",
      "ember-cli-dependency-checker": "^3.3.2",
      "ember-cli-htmlbars": "^6.3.0",
      "ember-cli-inject-live-reload": "^2.1.0",
      "ember-data": "~5.3.0",
      "ember-fetch": "^8.1.2",
      "ember-load-initializers": "^2.1.2",
      "ember-modifier": "^4.1.0",
      "ember-page-title": "^8.2.0",
      "ember-qunit": "^8.0.2",
      "ember-resolver": "^11.0.1",
      "ember-source": "~5.6.0",
      "ember-template-lint": "^5.13.0",
      "ember-welcome-page": "^7.0.2",
      "eslint": "^8.56.0",
      "eslint-config-prettier": "^9.1.0",
      "eslint-plugin-ember": "^11.12.0",
      "eslint-plugin-n": "^16.6.2",
      "eslint-plugin-prettier": "^5.1.3",
      "eslint-plugin-qunit": "^8.0.1",
      "loader.js": "^4.7.0",
      "prettier": "^3.2.4",
      "qunit": "^2.20.0",
      "qunit-dom": "^2.0.0",
      "stylelint": "^15.11.0",
      "stylelint-config-standard": "^34.0.0",
      "stylelint-prettier": "^4.1.0",
      "tracked-built-ins": "^3.3.0",
      "webpack": "^5.89.0"
    },
    "keywords": [],
    "license": "MIT",
    "name": "embroider-with-webpack-build-plugins-demo",
    "version": "0.0.0"
  },
  "meta": {
    "message": "Hello world!, from info.meta!"
  }
}
```

</details>